lib/rspec/mocks/argument_list_matcher.rb

```diff
@@ -46,16 +46,30 @@ def initialize(*expected_args)
         @expected_args = expected_args
         ensure_expected_args_valid!
       end
+      ruby2_keywords :initialize if Module.private_method_defined?(:ruby2_keywords)
 
       # @api public
-      # @param [Array] args
+      # @param [Array] actual_args
       #
       # Matches each element in the `expected_args` against the element in the same
       # position of the arguments passed to `new`.
       #
       # @see #initialize
-      def args_match?(*args)
-        Support::FuzzyMatcher.values_match?(resolve_expected_args_based_on(args), args)
+      def args_match?(*actual_args)
+        expected_args = resolve_expected_args_based_on(actual_args)
+
+        return false if expected_args.size != actual_args.size
+
+        if Hash.respond_to?(:ruby2_keywords_hash?, true)
```

Comment:
I think all we'd have to do is make this switch based on Ruby version (and/or optionally config)?